src/types.rs

```diff
@@ -114,12 +116,13 @@ impl FeeRate {
 
     /// Calculate absolute fee in Satoshis using size in weight units.
     pub fn fee_wu(&self, wu: usize) -> u64 {
-        self.fee_vb(wu.vbytes())
+        let vbytes = wu as f32 / 4.0;
+        (self.0 * vbytes).ceil() as u64
     }
 
     /// Calculate absolute fee in Satoshis using size in virtual bytes.
-    pub fn fee_vb(&self, vbytes: usize) -> u64 {
-        (self.as_sat_per_vb() * vbytes as f32).ceil() as u64
+    pub fn fee_vb(&self, vbytes: f32) -> u64 {
+        (self.0 * vbytes).ceil() as u64
```

Comment:
Why don't you use `self.as_sat_per_vb` instead of `self.0`? I feel it is more descriptive

Reply:
As `as_sat_per_vb` is an external interface, I personally find that `self.0` is more "direct". For someone reading the code, it is less steps to figure out what exactly is going on. Instead of going, *"What does `as_sat_per_vb` do? Oh it just returns `self.0`"*.